src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,110 @@
+write-once-read-many file system. The filesystem can list directory contents, check for a file's
+presence, and access the file by creating a file handle.
+
+The storage sources use the file system's put-object functionality to flush files into the cloud.
+The Tier Manager inside WiredTiger directs this flush. Only S3 storage source files pushed to the
+cloud are also copied to the local cache.
+
+@section extension_cloud_file_handle Cloud File Handle Interface
+
+Each storage source also implements the \c WT_FILE_HANDLE interface, to access the files
+on the file system. Since the object store is read-only, all the write access methods are disabled.
+
+@section extension_log_stat Messaging and Statistics
+The cloud provider SDKs use the term logging differently to WiredTiger. The SDK's logging definition
+refers to the error or informational messages that are generated when performing calls to the SDK.
+The SDKs also provide their own log levels which are translated into WiredTiger messaging levels.
+
+All of the cloud storage sources have their own messaging system and statistics to monitor the
+state of the SDKs and the storage sources. The messaging system captures error and informational
+messages from both the SDK and the custom storage source which are redirected into the WiredTiger
+messaging system. The statistics are printed when the WiredTiger verbosity is set to \c WT_VERBOSE
+or higher on the storage source.
+
+@section s3_store S3 Storage source
+The S3 store implements the following functionality defined by the storage source interface:
+
+- The bucket is expected to be a string argument comprised of the bucket name and the
+corresponding region separated by a ';', e.g. "bucket1;ap-southeast-2".
+
+- The S3 store handles authentication with AWS, requiring the user to provide an access token. The
+access token is comprised of an access key ID and a secret access key as a set. The S3 store
+requires the token to be provided as a single string argument of the access key followed by the
+secret key, separated by a ';'.
+
+- The S3 store configures a directory on the local file system as a cache for object files. The
+cache is used to keep a local copy of the file such that file operations doesn't need to go
+over the network, thus mitigating network latency.
+
+- When a S3 file handle is created, the local cache gets a copy of the S3 object if
+it doesn't already have it. Since the local cache is a directory on the WiredTiger database's file
+system, the file is accessed using a \c WT_FILE_HANDLE implementation native to WiredTiger.
+Hence, a \c WT_FILE_HANDLE is kept encapsulated in the S3 file handle.
+
+- The S3 store has created a logging class which extends the AWS SDK's logging system.
+
+@section azure_gcp_store Azure and GCP Storage Source
+The Azure and Google Cloud Platform (GCP) stores implement the following functionality:
+
+- The Azure and GCP stores follows different authentication methods to connect to their respective
```

Comment:
Grammar: "The Azure and GCP stores follows different" -> "The Azure and GCP stores follow different"